tsl/src/continuous_aggs/create.c

```diff
@@ -383,18 +384,21 @@ cagg_add_trigger_hypertable(Oid relid, char *trigarg)
 		foreach (cell, ht->data_nodes)
 		{
 			HypertableDataNode *node = lfirst(cell);
-			StringInfo command = makeStringInfo();
-			appendStringInfo(command,
-							 "CREATE TRIGGER %s AFTER INSERT OR UPDATE OR DELETE ON %s.%s FOR EACH "
-							 "ROW EXECUTE FUNCTION %s.%s(%d, %d)",
-							 quote_identifier(CAGGINVAL_TRIGGER_NAME),
-							 quote_identifier(NameStr(ht->fd.schema_name)),
-							 quote_identifier(NameStr(ht->fd.table_name)),
-							 quote_identifier(INTERNAL_SCHEMA_NAME),
-							 quote_identifier(CAGG_INVALIDATION_TRIGGER),
-							 node->fd.node_hypertable_id, /* distributed member hypertable ID */
-							 node->fd.hypertable_id /* Access Node hypertable ID */);
-			cmd_descr_data[i].sql = command->data;
+			int ret1, ret2;
+			char trigarg1[NAMEDATALEN], trigarg2[NAMEDATALEN];
+			CreateTrigStmt remote_stmt = stmt_template;
+
+			/* distributed member hypertable ID */
+			ret1 = snprintf(trigarg1, NAMEDATALEN, "%d", node->fd.node_hypertable_id);
+			/* Access Node hypertable ID */
+			ret2 = snprintf(trigarg2, NAMEDATALEN, "%d", node->fd.hypertable_id);
+			if (ret1 < 0 || ret1 >= NAMEDATALEN || ret2 < 0 || ret2 >= NAMEDATALEN)
+				ereport(ERROR,
+						(errcode(ERRCODE_INTERNAL_ERROR),
+						 errmsg("bad argument to continuous aggregate trigger")));
```

Comment:
The hypertable id is just an integer, so the error message in this case will never be triggered and an assert is probably better (but I think you don't need to use `snprintf` for this). Since you're just converting an integer to a string, `pg_ltoa` is probably a better case. Also, `trigarg1` and `trigarg2` is quite generic, which is why you have the comments below on what they are. Better to use something like `node_hypertable_str` and `hypertable_str`.
```suggestion
			pg_ltoa(node->fd.node_hypertable_id, trigarg1);
			pg_ltoa(node->fd.hypertable_id, trigarg2);
```

Reply:
Again copying the existing code here:
https://github.com/timescale/timescaledb/blob/2.5.0/tsl/src/continuous_aggs/create.c#L1878

Reply:
Should I change the existing code to have an `assert()`?

Reply:
This is just an integer, so I think `snprintf` is overkill and you should use `pg_ltoa`. This does not require an assert.

Reply:
I mean, should I remove the existing `ereport()` from the new code and the existing code?

Reply:
Actually, it seems reasonable to make that change as well. If you want to include that change in this PR I can review it.